### PR TITLE
test: fix profile recorder test

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?
Fix the error  in `trackRecorderSuite.TestHeapProfileRecorder` test which would fail in CI Test with a high probability

For 7b2a48c ,I run unit-test for 5 times. 4 pass ,1 failed for unrelational test.
For more details:
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36468/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36465/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36463/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36461/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36455/pipeline/

### Release note 
- None

### CODE REVIEW

File: util/profile/trackerrecorder_test.go [hunk 1]
```diff
@@ -61,20 +61,32 @@ func getRandomString(l int) string {
...
-	keys := make([]*mockCacheKey, 10000)
-	for i := 0; i < 10000; i++ {
+	keys := make([]*mockCacheKey, num)
+	for i := 0; i < 60000; i++ {
 		keys[i] = newMockHashKey(int64(i))
 		v := getRandomString(10)
 		lru.Put(keys[i], v)
 	}
+	// wait 10 sec to make runtime have scraped the heap profile sample
+	time.Sleep(10 * time.Second)
```
Comment (tiancaiamao) [inline util/profile/trackerrecorder_test.go:76]: We can't wait so long on CI ...

Comment (Yisaer) [reply util/profile/trackerrecorder_test.go]: removed

**Comment by Yisaer** [comment]
For 7b2a48c ,I run unit-test for 5 times. 4 pass ,1 failed for unrelational test.
For more details:
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36468/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36465/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36463/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36461/pipeline

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/36455/pipeline/

**Comment by tiancaiamao** [comment]
Fix https://github.com/pingcap/tidb/issues/17190

**Pull request merged**